py3status/modules/usbguard.py

```diff
@@ -0,0 +1,347 @@
+                device[x] = self.py3.composite_update(composite, {
+                    'index': '{}/{}'.format(device['id'], x.split('_')[-1])
+                })
+
+            new_device.append(self.py3.safe_format(self.format_device, device))
+
+        format_device_separator = self.py3.safe_format(
+            self.format_device_separator
+        )
+        format_device = self.py3.composite_join(
+            format_device_separator, new_device
+        )
+
+        return format_device, action
+
+    def _notify_user(self, device):
+        format_notification_message = self.py3.safe_format(
+            self.format_notification_message, device
+        )
+        format_notification_title = self.py3.safe_format(
+            self.format_notification_title, device
+        )
+        self.py3.notify_user(
+            msg=format_notification_message,
+            title=format_notification_title,
+            icon='/usr/share/icons/hicolor/scalable/apps/usbguard-icon.svg'
+        )
+
+    def usbguard(self):
+        self.devices = self._get_devices()
+        format_device, device_filter = self._manipulate_devices(self.devices)
+
+        format_button_filter = self.py3.safe_format(
+            self.format_button_filter, {'filter': device_filter}
+        )
+        self.py3.composite_update(
+            format_button_filter, {'index': 'button_filter'}
+        )
+        format_button_permanent = self.py3.safe_format(
+            self.format_button_permanent
+        )
+        self.py3.composite_update(
+            format_button_permanent, {'index': 'button_permanent'}
+        )
+
+        usbguard_data = {
+            'device': len(self.devices),
+            'format_button_filter': format_button_filter,
+            'format_button_permanent': format_button_permanent,
+            'format_device': format_device,
+        }
+
+        response = {
+            'cached_until': self.py3.CACHE_FOREVER,
+            'full_text': self.py3.safe_format(self.format, usbguard_data),
+        }
+
+        if self.allow_urgent:
+            response['urgent'] = True
+
+        return response
+
+    def kill(self):
+        self.killed.set()
+
+    def on_click(self, event):
+        index = event['index']
+        if isinstance(index, int):
+            return
+        elif index == 'button_filter':
+            if self.length > 1:
+                self.active_index += 1
+                if self.active_index >= self.length:
+                    self.active_index = 0
+        elif index == 'button_permanent':
+            self.permanent = not self.permanent
+        else:
+            device_id, policy_name = index.split('/')
+            device_id = int(device_id)
+
+            if self.init['notifications']:
+                for device in self.devices:
+                    if device['id'] == device_id:
+                        break
+                device['policy'] = policy_name
+                self._notify_user(device)
```

Comment:
>  Essentially, I don't think it can happen.

So you would rather a potential bug than adding 16 spaces and ensuring that there is not one :)

Race conditions are really hard to track down too

Reply:
>  Essentially, I don't think it can happen.

So you would rather a potential bug than adding 16 spaces and ensuring that there is not one :)

Race conditions are really hard to track down too